frontend/src/components/DetailsPage/HistoryList.tsx

```diff
@@ -1,45 +1,68 @@
 import HistoryListRow from "./HistoryListRow";
+import axios from "axios";
 
 type HistoryListProps = {
-    history: BpmnDiagramModel[]
     latestDiagram: BpmnDiagramModel
-    customDiagram: boolean
     setSnackbarOpen: Dispatch<SetStateAction<boolean>>
     setSnackbarMessage: Dispatch<SetStateAction<string>>
-    fetchHistory: () => void
     fetchDiagrams: () => void
     setTab: Dispatch<SetStateAction<string>>
 }
 
 function HistoryList(props: HistoryListProps) {
 
+    const [history, setHistory] = useState<BpmnDiagramModel[]>([])
+
+    const fetchHistory = () => {
+        axios.get("/api/bpmndiagrams/history/" + props.latestDiagram.businessKey)
+            .then(response => response.data)
+            .then(setHistory)
+    }
+
+    useEffect(fetchHistory, [props.latestDiagram.businessKey])
 
     return (
-        <Table size="small">
-            <TableHead>
-                <TableRow>
-                    <TableCell sx={{fontWeight: "bold"}}>Name</TableCell>
-                    <TableCell sx={{fontWeight: "bold"}} align="center">Filename</TableCell>
-                    <TableCell sx={{fontWeight: "bold"}} align="center">version</TableCell>
-                    {props.customDiagram &&
-                        <TableCell sx={{fontWeight: "bold"}} align="right">Delete</TableCell>
-                    }
-                </TableRow>
-            </TableHead>
-            <TableBody>
-                {props.history.map(diagram => <HistoryListRow
-                    key={diagram.id}
-                    diagram={diagram}
-                    fetchHistory={props.fetchHistory}
-                    fetchDiagrams={props.fetchDiagrams}
-                    latestDiagram={props.latestDiagram}
-                    setTab={props.setTab}
-                />)}
-            </TableBody>
-        </Table>
+        <CardContent>
+            <Typography variant="h5"
+                        color="secondary">History</Typography>
+            <Table size="small">
+                <TableHead>
+                    <TableRow>
+                        <TableCell>Name</TableCell>
+                        <TableCell align="center">Filename</TableCell>
+                        <TableCell align="center">Version</TableCell>
+                        <TableCell align="left">Comments</TableCell>
+                        {props.latestDiagram.customDiagram &&
+                            <TableCell align="right">Delete</TableCell>
+                        }
+                    </TableRow>
+                </TableHead>
+                <TableBody>
+                    {history
+                        .sort((diagram1, diagram2) => {
+                            if (diagram1.version > diagram2.version) {
+                                return -1
+                            } else return 1
+                        })
+                        .map(diagram => {
+                                if (diagram.id !== props.latestDiagram.id) {
```

Comment:
If I understand correctly this check fails only once in order to sort out the latestDiagram. Consider removing the latestDiagram from the list beforehand so this check is not needed anymore

Reply:
Thank you! Changed it, its much better this way